### PR TITLE
[WFU] Updated ARTopMenuViewController routing #trivial

### LINKED ISSUE #2304: [WFU] Push WFU vc as root
This will make sure the correct button is selected in the tab bar and the back button doesn't show up.

### PR DESCRIPTION
**Pull request by sarahscott**
This is a first working version of an update to the way ARTopMenuViewController does routing. It used to rely on the assumption that all root VCs were retained and organized by index, but this is no longer the case for both `ARFavoritesViewController` and `ARWorksForYouComponentViewController`.  The latter still indeed have corresponding indices but both are unretained and one carries parameters with is from `ARSwitchboard` that affect how it is rendered. 

There is of course an option to retain the `ARWorksForYouComponentViewController` and/or inject the `selectedArtist` after instantiation, but that can be discussed after routing is figured out. 

Fixes #2304 
Fixes https://github.com/artsy/emission/issues/490

### CODE REVIEW

File: Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m [hunk 1]
```diff
@@ -441,20 +432,60 @@ + (BOOL)shouldPresentViewControllerAsModal:(UIViewController *)viewController
...
+    
+    BOOL alreadySelectedTab = self.selectedTabIndex == index;
+    if (presentableController.viewControllers.count > 1) {
+        [presentableController popToRootViewControllerAnimated:(animated && alreadySelectedTab)];
+    }
+    
+    if (!alreadySelectedTab) {
+        if (index == NSNotFound) {
+            if ([viewController isKindOfClass:ARFavoritesViewController.class] || [viewController isKindOfClass:ARWorksForYouComponentViewController.class]) {
+                index = 4;
```
Comment (sarahscott) [inline Artsy/View_Controllers/App_Navigation/ARTopMenuViewController.m]: This of course won't be hard-coded and will have to figure out which index to use.

File: Artsy/Views/Utilities/ARTabContentView.m [hunk 2]
```diff
@@ -182,7 +187,7 @@ - (void)forceSetCurrentViewIndex:(NSInteger)index animated:(BOOL)animated
     }
 
     // Get the next View Controller, add to self
-    _currentNavigationController = [self navigationControllerForIndex:index];
+    _currentNavigationController = viewController;
```
Comment (sarahscott) [inline Artsy/Views/Utilities/ARTabContentView.m]: If this asks its data source for a view controller based on the index alone, all of the information from `ARSwitchboard` is lost (in this case, the selected artist).

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Podfile.lock [hunk 3]
```diff
@@ -419,4 +419,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: b498f78870627cf2b9bfba9e27bedd54a28d6439
 
-COCOAPODS: 1.2.1
+COCOAPODS: 1.2.0
```
Comment (alloy) [inline Podfile.lock]: Off-topic, but you should update your CP installation.

**Comment by alloy** [comment]
Looks good thus far 👍

**Comment by sarahscott** [comment]
Alright, ready for final review once CI is done!

**Comment by alloy** [comment]
Can you add a changelog please? That way we’ll know what areas to focus on more during QA.

**Comment by sarahscott** [comment]
Changelog entry ✅

**Pull request merged**